compiler/dyno/lib/uast/chpl-syntax-printer.cpp

```diff
@@ -0,0 +1,1065 @@
+      }
+      ss_ << "module ";
+      ss_ << node->name().c_str() << " ";
+      interpose(ss_, node->stmts(), "\n", "{", "}");
+    }
+
+    void visit(const MultiDecl* node) {
+      ss_ << "var ";
+
+      // TODO: Can this be generalized between TupleDecl and MultiDecl?
+      std::string delimiter = "";
+      for (auto decl : node->decls()) {
+        ss_ << delimiter;
+        ss_ << decl->toVariable()->name().c_str();
+        if (const Expression* te = decl->toVariable()->typeExpression()) {
+          ss_ << ": ";
+          printAst(ss_, te);
+        }
+        if (const Expression* ie = decl->toVariable()->initExpression()) {
+          ss_ << " = ";
+          printAst(ss_, ie);
+        }
+        delimiter = ", ";
+      }
+
+    }
+
+    void visit(const New* node) {
+      ss_ << "new ";
+      if (node->management() != New::Management::DEFAULT_MANAGEMENT) {
+        ss_ << kindToString(node->management()) << " ";
+      }
+      printAst(ss_, node->typeExpression());
+    }
+
+    void visit(const On* node) {
+      ss_ << "on ";
+      printAst(ss_, node->destination());
+      ss_ << " ";
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts(), "do ");
+    }
+
+    void visit(const OpCall* node) {
+      // TODO: Where do parens come in?
+      // ex: !(this && that)
+      // is different than !this && that
+      if (node->isUnaryOp()) {
+        ss_ << node->op().c_str();
+        assert(node->numActuals() == 1);
+        printAst(ss_, node->actual(0));
+
+      } else if (node->isBinaryOp()) {
+        assert(node->numActuals() == 2);
+        printAst(ss_, node->actual(0));
+        ss_ << " " << node->op().c_str() << " ";
+        printAst(ss_, node->actual(1));
+      }
+    }
+
+    void visit(const PrimCall* node) {
+      ss_ << "__primitive";
+      ss_ << "(";
+      ss_ << '"' << quoteStringForC(primTagToName(node->prim())) << '"' << ", ";
+      interpose(ss_, node->actuals(), ", ");
+      ss_ << ")";
+    }
+
+    void visit(const Range* node) {
+      if (auto lb = node->lowerBound()) {
+        printAst(ss_, lb);
+      }
+      ss_ << "..";
+      if (auto ub = node->upperBound()) {
+        printAst(ss_, ub);
+      }
+    }
+
+    void visit(const RealLiteral* node) { return visitLiteral(ss_, node); }
+
+    void visit(const Record* node) {
+      printLinkages(node);
+      ss_ << "record ";
+      ss_ << node->name().c_str() << " ";
+      interpose(ss_, node->decls(), "\n", "{", "}");
+    }
+
+    void visit(const Reduce* node) {
+      ss_ << node->op().c_str() << " ";
+      ss_ << "reduce ";
+      interpose(ss_, node->actuals(), ", ");
+    }
+
+    void visit(const Require* node) {
+      ss_ << "require ";
+      interpose(ss_, node->exprs(), ", ");
+    }
+
+    void visit(const Return* node) {
+      ss_ << "return";
+      if (node->value() != nullptr) {
+        ss_ << " ";
+        printAst(ss_, node->value());
+      }
+      ss_ << ";";
```

Comment:
IMO we should let whatever is driving the printing of statements (e.g. `printBlockWithStyle`) add the semicolons instead, so that we can use this for other things like return lifetimes.